File: p119e/pascal_triangle2.js
/**
 * @param {number} rowIndex
 * @return {number[]}
 */
var getRow = function(rowIndex) {
    if (rowIndex < 0) { return []; }
    if (rowIndex === 0) { return [1]; }
    let row = Array.from({length: 2}, (v) => 1); // rowIndex = 1
    for (let i = 2; i <= rowIndex; ++i) {
        let cur = Array.from({length: i + 1}, (v) => 1);
        for (let j = 1; j < i; ++j) {
            cur[j] = row[j - 1] + row[j];
        }
        row = cur;
    }
    return row;
};
// TEST
[
    3,5,7,10
].forEach((test) => {
    console.log("Pascal's triangle rowIndex of", test, "->",
                getRow(test));
});

File: p119e/pascal_triangle2.py
from typing import List


class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        row = [1]
        while rowIndex > 0:
            row = [0] + row
            for i in range(len(row) - 1):
                row[i] += row[i + 1]
            rowIndex -= 1
        return row


# TESTS
tests = [
    (0, [1]),
    (1, [1, 1]),
    (2, [1, 2, 1]),
    (3, [1, 3, 3, 1]),
    (4, [1, 4, 6, 4, 1]),
]
for t in tests:
    sol = Solution()
    actual = sol.getRow(t[0])
    print("Pascal triangle of", t[0], "index ->", actual)
    assert actual == t[1]
